app/attributes/attributecontroller.cpp

```diff
@@ -941,6 +941,78 @@ bool AttributeController::save()
   if ( !mFeatureLayerPair.layer() )
     return false;
 
+  const QStringList photoFields = QgsProject::instance()->entryList( QStringLiteral( "Mergin" ), QStringLiteral( "PhotoNaming/%1" ).arg( mFeatureLayerPair.layer()->id() ) );
+
+  QgsExpressionContext expressionContext = mFeatureLayerPair.layer()->createExpressionContext();
+  expressionContext << QgsExpressionContextUtils::formScope( mFeatureLayerPair.feature() );
+  if ( mVariablesManager )
+    expressionContext << mVariablesManager->positionScope();
+
+  expressionContext.setFields( mFeatureLayerPair.feature().fields() );
+  expressionContext.setFeature( mFeatureLayerPair.featureRef() );
+
+  // check for new photos
+  QMap<QUuid, std::shared_ptr<FormItem>>::iterator formItemsIterator = mFormItems.begin();
+  while ( formItemsIterator != mFormItems.end() )
+  {
+    std::shared_ptr<FormItem> item = formItemsIterator.value();
```

Comment:
Can we check here if the `item` has the photo name format defined and if not, just `continue`? We will save quite a few checks this way.

Ideally, we can reuse the `photoFields` for the check to avoid asking for QgsProject's instance and `readEntry` many times.
Something like:

```
  const QgsField field = item->field();
  if ( !photoFields.contains( field.name ) )
  {
    // increment index..
    continue;
  }
```